ios/engine/KMEI/KeymanEngine/Classes/InputViewController.swift

```diff
@@ -353,7 +359,8 @@ open class InputViewController: UIInputViewController, KeymanWebDelegate {
         }
       }
       
-      if textDocumentProxy.documentContextBeforeInput == nil {
+      if textDocumentProxy.documentContextBeforeInput == nil ||
+         (textDocumentProxy.documentContextBeforeInput == "\n" && Manager.shared.isSystemKeyboard) {
```

Comment:
Let's make sure we include a comment with an issue number on an esoteric condition like this that needs background to understand...